use_case_guide/real_time_dashboards.rst

```diff
@@ -138,141 +144,90 @@ and Citus will place matching shards on the same worker. This is called
+  CREATE OR REPLACE FUNCTION rollup_http_request() RETURNS void AS $$
+  DECLARE max_rollup_time TIMESTAMPTZ;
+  BEGIN
+    SELECT COALESCE(max(ingest_time), timestamp '10-10-1901')
+    FROM http_request_1min
+    INTO max_rollup_time;
+
+    INSERT INTO http_request_1min (
+      site_id, ingest_time, request_count,
+      success_count, error_count, average_response_time_msec
+    ) SELECT
+      site_id,
+      minute,
+      COUNT(1) as request_count,
+      SUM(CASE WHEN (status_code between 200 and 299) THEN 1 ELSE 0 END) as success_count,
+      SUM(CASE WHEN (status_code between 200 and 299) THEN 0 ELSE 1 END) as error_count,
+      SUM(response_time_msec) / COUNT(1) AS average_response_time_msec
+    FROM (
+      SELECT *,
+        date_trunc('minute', ingest_time) AS minute
+      FROM http_request
+    ) AS h
+    WHERE minute > max_rollup_time
+      AND minute < date_trunc('minute', now())
+    GROUP BY site_id, minute
+    ORDER BY minute ASC;
+  END;
+  $$ LANGUAGE plpgsql;
 
-   psql -tA -F" " -c "$QUERY" | xargs -P32 -n4 sh -c "$COMMAND"
+.. note::
 
-.. NOTE::
-
-  There are many ways to make sure the function is called periodically and no answer that
-  works well for every system. If you're able to run cron on the same machine as the
-  coordinator, and assuming you named the above script ``run_rollups.sh``, you can do something
-  as simple as this:
+  The above function should be called every minute. You could do this by
+  adding a crontab entry on the coordinator node:
 
   .. code-block:: bash
 
-     * * * * * /some/path/run_rollups.sh
+    * * * * * psql -c 'SELECT rollup_http_request();'
+
+  Alternately, an extension such as `pg_cron <https://github.com/citusdata/pg_cron>`_
+  allows you to schedule recurring queries directly from the database.
 
 The dashboard query from earlier is now a lot nicer:
 
 .. code-block:: sql
 
-  SELECT
-    request_count, success_count, error_count, average_response_time_msec
-  FROM http_request_1min
-  WHERE site_id = 1 AND date_trunc('minute', ingest_time) > date_trunc('minute', now()) - interval '5 minutes';
+  SELECT date_trunc('minute', ingest_time) as minute, request_count,
+         success_count, error_count, average_response_time_msec
+   FROM http_request_1min
+  WHERE site_id = 1
+    AND date_trunc('minute', ingest_time)
+        > date_trunc('minute', now()) - interval '5 minutes';
 
 Expiring Old Data
 -----------------
 
 The rollups make queries faster, but we still need to expire old data to avoid unbounded
 storage costs. Once you decide how long you'd like to keep data for each granularity, you
 could easily write a function to expire old data. In the following example, we decided to
-keep raw data for one day and 1-minute aggregations for one month.
+keep raw data for one day, and per-minute aggregations for one month.
 
 .. code-block:: plpgsql
 
-  -- another function for the coordinator
   CREATE OR REPLACE FUNCTION expire_old_request_data() RETURNS void
```

Comment:
Same note on whether we need to express this as a function. I think just showing a simple DELETE query would be nice, and then noting that in a production environment to cron this.